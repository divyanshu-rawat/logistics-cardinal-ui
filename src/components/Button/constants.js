export const SIZES = {
  XSMALL: 'xsmall',
  SMALL: 'small',
  DEFAULT: 'default',
  LARGE: 'large',
};

export const ACTION_TYPES = {
  DEFAULT: 'default',
  PRIMARY: 'primary',
  SUCCESS: 'success',
  INFO: 'info',
  WARNING: 'warning',
  DANGER: 'danger',
  LINK: 'link',
};
