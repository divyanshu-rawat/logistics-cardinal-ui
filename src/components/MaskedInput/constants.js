import createNumberMask from 'text-mask-addons/dist/createNumberMask';
import emailMask from 'text-mask-addons/dist/emailMask';

// DD.MM.YYYY
export const BIRTHDAY_MASK = [
  /\d/,
  /\d/,
  '.',
  /\d/,
  /\d/,
  '.',
  /\d/,
  /\d/,
  /\d/,
  /\d/,
];

// HH:mm
export const TIME_MASK = [/[0-9]/, /[0-9]/, ':', /[0-9]/, /[0-9]/];

// 10 % || - 10 %
export const PERCENTAGE_MASK = createNumberMask({
  prefix: '',
  includeThousandsSeparator: false,
  suffix: ' %',
  allowNegative: true,
});

// $ 10
export const MONEY_MASK = createNumberMask();

// rick@c137.com
export const EMAIL_MASK = emailMask;

// 10.29
export const DECIMAL_MASK = createNumberMask({
  prefix: '',
  allowDecimal: true,
  requireDecimal: true,
});

export default {
  BIRTHDAY_MASK,
  MONEY_MASK,
  EMAIL_MASK,
  DECIMAL_MASK,
  PERCENTAGE_MASK,
};
