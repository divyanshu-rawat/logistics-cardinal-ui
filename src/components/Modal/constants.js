export const modalSizes = {
  small: 'small',
  default: 'default',
};

export const modalClassName = 'CardinalModal__Content';
export const modalOverlayClassName = 'CardinalModal__Overlay';
