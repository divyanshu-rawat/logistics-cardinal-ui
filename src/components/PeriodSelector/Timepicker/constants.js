const theme = {
  brandPrimary: '#3f9bfd',
  brandPrimaryLight: '#375f8a',
  greyMedium: '#343a42',
  greyDeep: '#292d33',
  greyLight: '#3f454d',
  inactive: '#96a2b0',
};

export default theme;
