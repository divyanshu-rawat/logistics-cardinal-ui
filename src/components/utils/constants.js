export const SIZES = {
  lg: 'lg',
  md: 'md',
  sm: 'sm',
  xs: 'xs',
};
