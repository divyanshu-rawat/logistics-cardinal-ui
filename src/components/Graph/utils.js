import { SIZES } from '../utils/constants';

/**
 * Generate random shifts data
 * @returns {{shifts: number}[]}
 */
export function generateRandomShifts() {
  const MONTHS = [
    'Jan',
    'Feb',
    'Mar',
    'Apr',
    'May',
    'Jun',
    'Jul',
    'Aug',
    'Sep',
    'Oct',
    'Nov',
    'Dec',
  ];
  return [...new Array(12)].map((_, index) => ({
    shifts: Math.round(Math.random() * 50),
    month: MONTHS[index],
  }));
}

export function getStrokeWidth(size) {
  let strokeSize;

  switch (size) {
    case SIZES.lg:
      strokeSize = 4;
      break;
    case SIZES.md:
      strokeSize = 3;
      break;
    case SIZES.sm:
      strokeSize = 2;
      break;
    default:
      strokeSize = 1;
  }

  return strokeSize;
}
