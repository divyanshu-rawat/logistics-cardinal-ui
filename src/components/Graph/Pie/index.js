import React from 'react';
import PropTypes from 'prop-types';
import { Pie as RCPie } from 'recharts';

import { SIZES } from '../../utils/constants';

class Pie extends RCPie {
  static propTypes = {
    ...RCPie.propTypes,
    /**
     * Size of the inner radius
     */
    innerRadiusSize: PropTypes.oneOf(Object.values(SIZES)).isRequired,
    /**
     * Size of the outer radius
     */
    outerRadiusSize: PropTypes.oneOf(Object.values(SIZES)).isRequired,
    /**
     * Data that contains an array of objects
     */
    data: PropTypes.array.isRequired,
    /**
     * Key of the objects
     */
    dataKey: PropTypes.string.isRequired,
    /**
     * Children which contains Cell components
     */
    children: PropTypes.node.isRequired,
  };

  static defaultProps = {
    ...RCPie.defaultProps,
    innerRadiusSize: 'sm',
    outerRadiusSize: 'lg',
  };

  static getRadiusSize = (size) => {
    let percentage;

    switch (size) {
      case SIZES.lg:
        percentage = '100%';
        break;
      case SIZES.md:
        percentage = '85%';
        break;
      case SIZES.sm:
        percentage = '70%';
        break;
      default:
        percentage = '55%';
    }

    return percentage;
  };

  render() {
    const { data, dataKey, children, ...props } = this.props;

    const innerRadiusSize = Pie.getRadiusSize(props.innerRadiusSize);
    const outerRadiusSize = Pie.getRadiusSize(props.outerRadiusSize);

    return (
      <RCPie
        data={data}
        dataKey={dataKey}
        innerRadius={innerRadiusSize}
        outerRadius={outerRadiusSize}
        {...props}
      >
        {children}
      </RCPie>
    );
  }
}

export default Pie;
