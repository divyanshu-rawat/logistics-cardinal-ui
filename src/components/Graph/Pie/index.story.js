import React from 'react';
import { storiesOf } from '@storybook/react';
import { withInfo } from '@storybook/addon-info';
import { select } from '@storybook/addon-knobs';
import { Cell } from 'recharts';
import withTests from '../../../utils/withTests';
import { GRAPH } from '../../../../.storybook/sections';

import PieChart from '../PieChart';
import Pie from '.';
import { SIZES } from '../../utils/constants';

storiesOf(`${GRAPH}Pie`, module)
  .addDecorator(withTests('Pie/tests/index'))
  .add(
    'Simple',
    withInfo()(() => (
      <PieChart width={400} height={400}>
        <Pie
          data={[{ value: 30 }, { value: 20 }, { value: 15 }]}
          dataKey="value"
          innerRadiusSize={select(
            'innerRadiusSize',
            Object.values(SIZES),
            SIZES.xs,
          )}
          outerRadiusSize={select(
            'outerRadiusSize',
            Object.values(SIZES),
            SIZES.lg,
          )}
        >
          <Cell key="CellOne" fill="rgba(79, 127, 177, 0.3)" stroke="#4f7fb1" />
          <Cell key="CellTwo" fill="rgba(109,179,36,0.3)" stroke="#6DB324" />
          <Cell key="CellThree" fill="#662147" stroke="#E6499F" />
        </Pie>
      </PieChart>
    )),
  );
