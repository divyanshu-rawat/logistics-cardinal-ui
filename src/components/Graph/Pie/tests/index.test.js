import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

import Pie from '..';
import { SIZES } from '../../../utils/constants';

const renderedComponent = () =>
  shallow(
    <Pie
      data={[]}
      dataKey="value"
      innerRadiusSize={SIZES.xs}
      outerRadiusSize={SIZES.lg}
    >
      Awesome Pie Chart
    </Pie>,
  );

describe('<Pie />', () => {
  it('should render the component', () => {
    const wrapper = renderedComponent();
    const tree = toJson(wrapper);
    expect(tree).toMatchSnapshot();
  });
});
