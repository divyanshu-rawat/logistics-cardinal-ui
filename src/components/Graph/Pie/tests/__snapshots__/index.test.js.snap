// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Pie /> should render the component 1`] = `
<Pie
  animationBegin={400}
  animationDuration={1500}
  animationEasing="ease"
  cx="50%"
  cy="50%"
  data={Array []}
  dataKey="value"
  endAngle={360}
  fill="#808080"
  hide={false}
  innerRadius={0}
  innerRadiusSize="xs"
  isAnimationActive={true}
  labelLine={true}
  legendType="rect"
  minAngle={0}
  nameKey="name"
  outerRadius="80%"
  outerRadiusSize="lg"
  paddingAngle={0}
  startAngle={0}
  stroke="#fff"
>
  Awesome Pie Chart
</Pie>
`;
